torch/_subclasses/fake_tensor.py

```diff
@@ -1412,6 +1434,10 @@ def _validate_cache_key(
         # Avoid caching for any ops that would require a more sophisticated
         # caching implementation, e.g., data dependent ops or ops that modify
         # the inputs.
+
+        if isinstance(func, torch._ops.HigherOrderOperator):
+            return
```

Comment:
Hmm

Reply:
This is interesting. What do we do if we capture a subgraph that has a data-dependent operation (like nonzero) in it? In that case, we want to either bypass the cache or error out

Reply:
Thats a good point! For now, I have restricted it to invoke_subgraph and that too only when identifier is set (which hopefully should be fine because Dynamo would test for those ops).

Reply:
And other thing we should do is that Dynamo should enforce this while preparing the invoke_subgraph 

```
![image](https://github.com/user-attachments/assets/5258f71d-f6c2-45c2-8ba7-bd5212949e46)

```

Reply:
yup